app/controllers/cards_controller.rb

```diff
@@ -0,0 +1,64 @@
+class CardsController < ApplicationController
```

Comment:
You don't appear to have any sort of protection against users performing actions against cards they shouldn't have access to. I would watch out for that whenever possible.